src/main/java/bio/overture/ego/controller/AuthController.java

```diff
@@ -128,22 +128,32 @@ public ResponseEntity<String> user(
 
     val user = (CustomOAuth2User) authentication.getPrincipal();
 
-    val passportJwtToken = (authentication.getAuthorizedClientRegistrationId().equals(PASSPORT_CLIENT_NAME)) ?
-        passportService.getPassportToken(((CustomOAuth2User) authentication.getPrincipal()).getAccessToken()) :
+
+    val passportJwtToken = (user.getClaim(GA4GH_PASSPORT_SCOPE) != null) ?
+        passportService.getPassportToken(
+            authentication.getAuthorizedClientRegistrationId(),
+            user.getAccessToken()) :
         null;
 
+    Optional<ProviderType> providerType = ProviderType
+        .findIfExist(authentication.getAuthorizedClientRegistrationId());
+
+    if(user.getClaim(GA4GH_PASSPORT_SCOPE) != null && providerType.isEmpty()){
+      providerType = Optional.of(ProviderType.PASSPORT);
+    }
```

Comment:
Assigning a generic providerType "PASSPORT" to any passport provider that contains the scope `ga4gh_passport_v1`